### PR TITLE
[flutter tools] Revert desktop device name changes and print the category instead

### PR DESCRIPTION
**Pull request by jamesderlin**
Revert #58812 and instead make `flutter devices`/`flutter doctor`
print the device category, which is what the IntelliJ IDE plugin
does.  The format will be:

    • deviceName (category) • deviceID • targetPlatformName • osName

This does mean that the device name again will be the same the OS
name for desktop devices.

@devoncarew also suggested suffixing `-device` for the device names.

## Related Issues

#58812

## Tests

None

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

I reverted tests back to the state prior to #58812.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/lib/src/linux/linux_device.dart [hunk 1]
```diff
@@ -17,7 +17,7 @@ import 'linux_workflow.dart';
 /// A device that represents a desktop Linux target.
 class LinuxDevice extends DesktopDevice {
   LinuxDevice() : super(
-      'linux',
+      'linux-device',
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/linux/linux_device.dart:20]: `linux-desktop`?

File: packages/flutter_tools/lib/src/macos/macos_device.dart [hunk 2]
```diff
@@ -15,7 +15,7 @@ import 'macos_workflow.dart';
 /// A device that represents a desktop MacOS target.
 class MacOSDevice extends DesktopDevice {
   MacOSDevice() : super(
-      'macos',
+      'macos-device',
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/macos/macos_device.dart:18]: `macos-desktop`?

File: packages/flutter_tools/lib/src/windows/windows_device.dart [hunk 3]
```diff
@@ -18,7 +18,7 @@ import 'windows_workflow.dart';
 /// A device that represents a desktop Windows target.
 class WindowsDevice extends DesktopDevice {
   WindowsDevice() : super(
-      'windows',
+      'windows-device',
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/windows/windows_device.dart:21]: `windows-desktop`?

**Review by devoncarew — COMMENTED** [review]
Thanks for the changes!

One main comment - I think the device ID for the desktop devices should have a `-desktop` suffix (instead of a `-device` suffix) to help differentiate them from other devices.

**Comment by stuartmorgan** [comment]
> to help differentiate them from other devices

Out of curiosity, what devices are `windows`, `macos`, and `linux` being confused with? I don't really mind suffixing them as long as the unsuffixed versions still work, but I'm not sure what problem is being solved here.

File: packages/flutter_tools/lib/src/device.dart [hunk 4]
```diff
@@ -559,7 +559,7 @@ abstract class Device {
         supportIndicator += ' ($type)';
       }
       table.add(<String>[
-        device.name,
```
Comment (stuartmorgan) [inline packages/flutter_tools/lib/src/device.dart:562]: This will change everything, not just desktop; is that desired?

Comment (stuartmorgan) [inline packages/flutter_tools/lib/src/device.dart:562]: This change doesn't seem to have any associated test changes. If we want this behavior change, we should test it.

**Comment by devoncarew** [comment]
> > to help differentiate them from other devices
> 
> Out of curiosity, what devices are `windows`, `macos`, and `linux` being confused with? I don't really mind suffixing them as long as the unsuffixed versions still work, but I'm not sure what problem is being solved here.

I believe the main motivation is to avoid the `Linux • Linux • linux-x64 • Linux` devices description (from https://github.com/flutter/flutter/pull/58812).

**Comment by stuartmorgan** [comment]
> I believe the main motivation is to avoid the `Linux • Linux • linux-x64 • Linux` devices description (from #58812).

That isn't the line now, so mission accomplished if that's the whole problem statement 🙂  But the statement there was "it has confused people", which doesn't really tell me what the specific problem is. What, exactly, is it being confused with, or confusing them about? Do we know enough to know whether "Linux (desktop) • linux • linux-x64 • Linux" will still be confusing, and how? In what way will adding the word "desktop" to that line a second time reduce confusion?

Without understanding exactly what problem we're trying to solve, it's hard to evaluate proposals.

For instance, the problem is that it's just the word Linux a bunch of times and people don't know why it keeps saying the same thing, adding more details to the last component, the way we do on other platforms (e.g., `Mac OS X 10.15.5 19F101`), would address that in a way that's useful rather than just repeating a different word.

**Comment by jonahwilliams** [comment]
I agree with @stuartmorgan that `foo-desktop` is a bit redundant. The category option should be enough to break up the text, but we should be including more details in the OS name category anyway. And as always, it should be tested - even if it wasn't being tested before

Eventually, we'll also fix the device architecture to something like `x64/`arm64`/`arm32` which should reduce more of the repetition.

**Comment by jonahwilliams** [comment]
Actually it was tested 😄 

```
02:11 +239: test/commands.shard/hermetic/devices_test.dart: devices available devices and diagnostics                                                                                                  
Expected: '2 connected devices:\n'
            '\n'
            'ephemeral • ephemeral • android-arm    • Test SDK (1.2.3) (emulator)\n'
            'webby     • webby     • web-javascript • Web SDK (1.2.4) (emulator)\n'
            '\n'
            '• Cannot connect to device ABC\n'
            ''
  Actual: '2 connected devices:\n'
            '\n'
            'ephemeral (mobile) • ephemeral • android-arm    • Test SDK (1.2.3) (emulator)\n'
            'webby (mobile)     • webby     • web-javascript • Web SDK (1.2.4) (emulator)\n'
            '\n'
            '• Cannot connect to device ABC\n'
            ''
   Which: is different.
          Expected: ... ephemeral • ephemera ...
            Actual: ... ephemeral (mobile) • ...
                                  ^
           Differ at offset 34
```

**Comment by devoncarew** [comment]
> Without understanding exactly what problem we're trying to solve, it's hard to evaluate proposals.

Ah; the main overall issue is that the original change regressed the UX in IDEs. Whatever else we change, we should address that. You can see the issue in the linked PR. I care much less about reducing redundancy in the `flutter devices` output (or rather, it's much less of an issue).

**Comment by stuartmorgan** [comment]
> Ah; the main overall issue is that the original change regressed the UX in IDEs. Whatever else we change, we should address that. You can see the issue in the linked PR.

I understand that part; I'm asking for more specific details about the motivation for the initial PR. The change from including "desktop" in the name to displaying the category in `devices` addressed the regression entirely unless I'm missing something.

> I care much less about reducing redundancy in the `flutter devices` output (or rather, it's much less of an issue).

This PR does two seemingly unrelated things: fixes the regression for IDE display (I'm all for that), and then for reasons I don't understand *adds* redundancy to the `devices` display. It's the second part I'm trying to understand the motivation for.

File: packages/flutter_tools/lib/src/device.dart (hunk 4 above)
Comment (jamesderlin) [reply packages/flutter_tools/lib/src/device.dart:562]: That was *my* intent, yes.  Whether it's desirable it up to review. ;)

**Comment by jamesderlin** [comment]
> > to help differentiate them from other devices
> 
> Out of curiosity, what devices are `windows`, `macos`, and `linux` being confused with? I don't really mind suffixing them as long as the unsuffixed versions still work, but I'm not sure what problem is being solved here.

macos: Nothing.

windows: In practice, nothing.  Historically there have been other Windows devices other than desktops.  Granted, those other Windows-based platforms are dead now, and Flutter doesn't support them.

linux: This one is the confusing one, and I changed the macos and windows devices for consistency.  Android is Linux-based; does that count?  What about other potential Linux-based platforms?  It was particularly confusing if you ran `flutter doctor` without any explicitly attached ephemeral devices and you saw "Linux • Linux • linux-x64 • Linux" listed.

I'm not particularly tied to changing the device IDs (which is why I settled with just lowercasing them in #58812), so I don't mind leaving that alone, but I thought it would be nice to break up the device-name/device-id/os-name monotony slightly.

**Comment by jamesderlin** [comment]
I fixed the tests, and since the device ID changes seem a bit controversial, I've changed those back

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Pull request merged**